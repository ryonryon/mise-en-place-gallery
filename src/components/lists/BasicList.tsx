import styled from "@emotion/styled";
import * as React from "react";

import { secondary } from "../../constants/colors";

const StyledListDiv = styled.div`
  margin: 16px;
  height: 300px;
  width: 90%;
  overflow: scroll;
  color: ${secondary};

  ul {
    padding: 0;
  }

  li {
    list-style: none;
    padding: 8px;
    margin: 8px;

    &:hover {
      background-color: rgba(221, 221, 221, 0.4);

      cursor: pointer;
    }
  }
`;

interface Props {
  list: string[];
}

const BasicList = ({ list }: Props) => (
  <StyledListDiv>
    <ul>
      {list.map((item: string, index: number) => (
        <li key={index}>{item}</li>
      ))}
    </ul>
  </StyledListDiv>
);

export default BasicList;
