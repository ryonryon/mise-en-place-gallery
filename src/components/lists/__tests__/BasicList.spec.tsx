import * as React from "react";
import * as renderer from "react-test-renderer";

import BasicList from "../BasicList";

describe("BasicList test", () => {
  it("tests BasicList test render correctly", () => {
    const basicList = renderer.create(<BasicList list={["list item1", "list item2", "list item3"]} />).toJSON();

    expect(basicList).toMatchSnapshot();
  });
});
