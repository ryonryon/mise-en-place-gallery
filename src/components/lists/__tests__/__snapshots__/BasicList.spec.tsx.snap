// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicList test tests BasicList test render correctly 1`] = `
<div
  className="css-cm3tjs"
>
  <ul>
    <li>
      list item1
    </li>
    <li>
      list item2
    </li>
    <li>
      list item3
    </li>
  </ul>
</div>
`;
